### ISSUE TITLE
[no-unused-vars] Incorrect code location when self referencing

### ISSUE THREAD

**Issue by Zodiase** [typescript-eslint/typescript-eslint#2353]
**Repro**

![image](https://user-images.githubusercontent.com/1815174/89209393-0c261880-d573-11ea-8bf5-416feae013ff.png)

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": ['warn', { ignoreRestSiblings: true }]
  }
}
```

```TS
const foobar = () => ({
    getFoobar: () => foobar() //<-- 'foobar' is assigned a value but never used.
});
```

**Expected Result**
`foobar` on line 1 is labeled as not used.

**Actual Result**
`foobar()` on line 2 is labeled instead.

**Additional Info**

```
eslint:cli CLI args: [ 'src/test.ts', '--debug' ] +0ms
  eslint:cli Running on files +4ms
  eslint:config-array-factory Loading JSON config file: package.json +0ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components', loose: false } ] +0ms
  eslint:ignore-pattern   processed: { basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components', patterns: [ '/**/node_modules/*' ] } +1ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components', loose: false } ] +0ms
  eslint:ignore-pattern   processed: { basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components', patterns: [ '/**/node_modules/*' ] } +1ms
  eslint:file-enumerator Start to iterate files: [ 'src/test.ts' ] +0ms
  eslint:file-enumerator File: src/test.ts +1ms
  eslint:cascading-config-array-factory Load config files for src. +0ms
  eslint:cascading-config-array-factory No cache found: src. +0ms
  eslint:config-array-factory Config file not found on src +5ms
  eslint:cascading-config-array-factory No cache found: /Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components. +0ms
  eslint:config-array-factory Loading JS config file: .eslintrc.js +0ms
  eslint:config-array-factory Config file found: .eslintrc.js +1ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/recommended"} relative to .eslintrc.js +1ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from .eslintrc.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@3.7.1 (node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +1ms
  eslint:config-array-factory Plugin node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 249ms +249ms
  eslint:config-array-factory Loading {extends:"./configs/base"} relative to node_modules/@typescript-eslint/eslint-plugin/dist/index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module './configs/base/package.json'
Require stack:
- node_modules/@typescript-eslint/eslint-plugin/dist/index.js +0ms
  eslint:config-array-factory Loaded: ./configs/base (node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.js) +0ms
  eslint:config-array-factory Loading JS config file: node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.js +0ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.js +1ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@3.7.1 (node_modules/@typescript-eslint/parser/dist/index.js) +1ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.js +50ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@3.7.1 (node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +0ms
  eslint:config-array-factory Plugin node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"./configs/eslint-recommended"} relative to node_modules/@typescript-eslint/eslint-plugin/dist/index.js +0ms
  eslint:config-array-factory package.json was not found: Cannot find module './configs/eslint-recommended/package.json'
Require stack:
- node_modules/@typescript-eslint/eslint-plugin/dist/index.js +0ms
  eslint:config-array-factory Loaded: ./configs/eslint-recommended (node_modules/@typescript-eslint/eslint-plugin/dist/configs/eslint-recommended.js) +0ms
  eslint:config-array-factory Loading JS config file: node_modules/@typescript-eslint/eslint-plugin/dist/configs/eslint-recommended.js +0ms
  eslint:config-array-factory Loading {extends:"plugin:react/recommended"} relative to .eslintrc.js +2ms
  eslint:config-array-factory Loading plugin "react" from .eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-react@7.20.5 (node_modules/eslint-plugin-react/index.js) +0ms
  eslint:config-array-factory Plugin node_modules/eslint-plugin-react/index.js loaded in: 78ms +78ms
  eslint:config-array-factory Loading plugin "react" from node_modules/eslint-plugin-react/index.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-react@7.20.5 (node_modules/eslint-plugin-react/index.js) +1ms
  eslint:config-array-factory Plugin node_modules/eslint-plugin-react/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"plugin:react-hooks/recommended"} relative to .eslintrc.js +0ms
  eslint:config-array-factory Loading plugin "react-hooks" from .eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-react-hooks@4.0.8 (node_modules/eslint-plugin-react-hooks/index.js) +0ms
  eslint:config-array-factory Plugin node_modules/eslint-plugin-react-hooks/index.js loaded in: 1ms +1ms
  eslint:config-array-factory Loading plugin "react-hooks" from node_modules/eslint-plugin-react-hooks/index.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-react-hooks@4.0.8 (node_modules/eslint-plugin-react-hooks/index.js) +0ms
  eslint:config-array-factory Plugin node_modules/eslint-plugin-react-hooks/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"prettier"} relative to .eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-config-prettier@6.11.0 (node_modules/eslint-config-prettier/index.js) +0ms
  eslint:config-array-factory Loading JS config file: node_modules/eslint-config-prettier/index.js +0ms
  eslint:config-array-factory Loading {extends:"prettier/@typescript-eslint"} relative to .eslintrc.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module 'eslint-config-prettier/@typescript-eslint/package.json'
Require stack:
- .eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-config-prettier/@typescript-eslint (node_modules/eslint-config-prettier/@typescript-eslint.js) +0ms
  eslint:config-array-factory Loading JS config file: node_modules/eslint-config-prettier/@typescript-eslint.js +0ms
  eslint:config-array-factory Loading {extends:"prettier/babel"} relative to .eslintrc.js +0ms
  eslint:config-array-factory package.json was not found: Cannot find module 'eslint-config-prettier/babel/package.json'
Require stack:
- .eslintrc.js +1ms
  eslint:config-array-factory Loaded: eslint-config-prettier/babel (node_modules/eslint-config-prettier/babel.js) +0ms
  eslint:config-array-factory Loading JS config file: node_modules/eslint-config-prettier/babel.js +0ms
  eslint:config-array-factory Loading {extends:"prettier/react"} relative to .eslintrc.js +0ms
  eslint:config-array-factory package.json was not found: Cannot find module 'eslint-config-prettier/react/package.json'
Require stack:
- .eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-config-prettier/react (node_modules/eslint-config-prettier/react.js) +0ms
  eslint:config-array-factory Loading JS config file: node_modules/eslint-config-prettier/react.js +0ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from .eslintrc.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@3.7.1 (node_modules/@typescript-eslint/parser/dist/index.js) +1ms
  eslint:config-array-factory Loading plugin "react" from .eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-react@7.20.5 (node_modules/eslint-plugin-react/index.js) +0ms
  eslint:config-array-factory Plugin node_modules/eslint-plugin-react/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading plugin "react-hooks" from .eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-react-hooks@4.0.8 (node_modules/eslint-plugin-react-hooks/index.js) +0ms
  eslint:config-array-factory Plugin node_modules/eslint-plugin-react-hooks/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from .eslintrc.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@3.7.1 (node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +0ms
  eslint:config-array-factory Plugin node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 0ms +0ms
  eslint:cascading-config-array-factory No cache found: /Users/xh/Workspace/development-docker-setup/webapp/gauss/app. +390ms
  eslint:config-array-factory Config file not found on /Users/xh/Workspace/development-docker-setup/webapp/gauss/app +1ms
  eslint:cascading-config-array-factory No cache found: /Users/xh/Workspace/development-docker-setup/webapp/gauss. +1ms
  eslint:config-array-factory Loading package.json config file: /Users/xh/Workspace/development-docker-setup/webapp/gauss/package.json +0ms
  eslint:config-array-factory Loading JSON config file: /Users/xh/Workspace/development-docker-setup/webapp/gauss/package.json +0ms
  eslint:config-array-factory Error reading package.json file: /Users/xh/Workspace/development-docker-setup/webapp/gauss/package.json +0ms
  eslint:config-array-factory Config file not found on /Users/xh/Workspace/development-docker-setup/webapp/gauss +0ms
  eslint:cascading-config-array-factory No cache found: /Users/xh/Workspace/development-docker-setup/webapp. +0ms
  eslint:config-array-factory Config file not found on /Users/xh/Workspace/development-docker-setup/webapp +1ms
  eslint:cascading-config-array-factory No cache found: /Users/xh/Workspace/development-docker-setup. +1ms
  eslint:config-array-factory Loading package.json config file: /Users/xh/Workspace/development-docker-setup/package.json +1ms
  eslint:config-array-factory Loading JSON config file: /Users/xh/Workspace/development-docker-setup/package.json +0ms
  eslint:config-array-factory Error reading package.json file: /Users/xh/Workspace/development-docker-setup/package.json +0ms
  eslint:config-array-factory Config file not found on /Users/xh/Workspace/development-docker-setup +0ms
  eslint:cascading-config-array-factory No cache found: /Users/xh/Workspace. +1ms
  eslint:config-array-factory Config file not found on /Users/xh/Workspace +0ms
  eslint:cascading-config-array-factory No cache found: /Users/xh. +0ms
  eslint:cascading-config-array-factory Stop traversing because of considered root. +0ms
  eslint:rules Loading rule 'constructor-super' (remaining=281) +0ms
  eslint:rules Loading rule 'getter-return' (remaining=280) +0ms
  eslint:rules Loading rule 'no-const-assign' (remaining=279) +1ms
  eslint:rules Loading rule 'no-dupe-args' (remaining=278) +0ms
  eslint:rules Loading rule 'no-dupe-class-members' (remaining=277) +0ms
  eslint:rules Loading rule 'no-dupe-keys' (remaining=276) +0ms
  eslint:rules Loading rule 'no-func-assign' (remaining=275) +1ms
  eslint:rules Loading rule 'no-import-assign' (remaining=274) +0ms
  eslint:rules Loading rule 'no-new-symbol' (remaining=273) +0ms
  eslint:rules Loading rule 'no-obj-calls' (remaining=272) +0ms
  eslint:rules Loading rule 'no-redeclare' (remaining=271) +1ms
  eslint:rules Loading rule 'no-setter-return' (remaining=270) +0ms
  eslint:rules Loading rule 'no-this-before-super' (remaining=269) +0ms
  eslint:rules Loading rule 'no-undef' (remaining=268) +0ms
  eslint:rules Loading rule 'no-unreachable' (remaining=267) +1ms
  eslint:rules Loading rule 'no-unsafe-negation' (remaining=266) +0ms
  eslint:rules Loading rule 'no-var' (remaining=265) +0ms
  eslint:rules Loading rule 'prefer-const' (remaining=264) +1ms
  eslint:rules Loading rule 'prefer-rest-params' (remaining=263) +2ms
  eslint:rules Loading rule 'prefer-spread' (remaining=262) +0ms
  eslint:rules Loading rule 'valid-typeof' (remaining=261) +1ms
  eslint:rules Loading rule 'no-array-constructor' (remaining=260) +4ms
  eslint:rules Loading rule 'no-empty-function' (remaining=259) +1ms
  eslint:rules Loading rule 'no-extra-semi' (remaining=258) +2ms
  eslint:rules Loading rule 'no-unused-vars' (remaining=257) +2ms
  eslint:rules Loading rule 'arrow-body-style' (remaining=256) +9ms
  eslint:rules Loading rule 'curly' (remaining=255) +0ms
  eslint:rules Loading rule 'lines-around-comment' (remaining=254) +1ms
  eslint:rules Loading rule 'max-len' (remaining=253) +0ms
  eslint:rules Loading rule 'no-confusing-arrow' (remaining=252) +0ms
  eslint:rules Loading rule 'no-mixed-operators' (remaining=251) +1ms
  eslint:rules Loading rule 'no-tabs' (remaining=250) +0ms
  eslint:rules Loading rule 'no-unexpected-multiline' (remaining=249) +0ms
  eslint:rules Loading rule 'prefer-arrow-callback' (remaining=248) +1ms
  eslint:rules Loading rule 'quotes' (remaining=247) +0ms
  eslint:rules Loading rule 'array-bracket-newline' (remaining=246) +0ms
  eslint:rules Loading rule 'array-bracket-spacing' (remaining=245) +0ms
  eslint:rules Loading rule 'array-element-newline' (remaining=244) +1ms
  eslint:rules Loading rule 'arrow-parens' (remaining=243) +0ms
  eslint:rules Loading rule 'arrow-spacing' (remaining=242) +0ms
  eslint:rules Loading rule 'block-spacing' (remaining=241) +1ms
  eslint:rules Loading rule 'brace-style' (remaining=240) +0ms
  eslint:rules Loading rule 'comma-dangle' (remaining=239) +0ms
  eslint:rules Loading rule 'comma-spacing' (remaining=238) +0ms
  eslint:rules Loading rule 'comma-style' (remaining=237) +1ms
  eslint:rules Loading rule 'computed-property-spacing' (remaining=236) +0ms
  eslint:rules Loading rule 'dot-location' (remaining=235) +0ms
  eslint:rules Loading rule 'eol-last' (remaining=234) +0ms
  eslint:rules Loading rule 'func-call-spacing' (remaining=233) +1ms
  eslint:rules Loading rule 'function-call-argument-newline' (remaining=232) +0ms
  eslint:rules Loading rule 'function-paren-newline' (remaining=231) +0ms
  eslint:rules Loading rule 'generator-star-spacing' (remaining=230) +1ms
  eslint:rules Loading rule 'implicit-arrow-linebreak' (remaining=229) +0ms
  eslint:rules Loading rule 'indent' (remaining=228) +0ms
  eslint:rules Loading rule 'jsx-quotes' (remaining=227) +0ms
  eslint:rules Loading rule 'key-spacing' (remaining=226) +0ms
  eslint:rules Loading rule 'keyword-spacing' (remaining=225) +1ms
  eslint:rules Loading rule 'linebreak-style' (remaining=224) +0ms
  eslint:rules Loading rule 'multiline-ternary' (remaining=223) +0ms
  eslint:rules Loading rule 'newline-per-chained-call' (remaining=222) +0ms
  eslint:rules Loading rule 'new-parens' (remaining=221) +1ms
  eslint:rules Loading rule 'no-extra-parens' (remaining=220) +0ms
  eslint:rules Loading rule 'no-floating-decimal' (remaining=219) +0ms
  eslint:rules Loading rule 'no-mixed-spaces-and-tabs' (remaining=218) +0ms
  eslint:rules Loading rule 'no-multi-spaces' (remaining=217) +1ms
  eslint:rules Loading rule 'no-multiple-empty-lines' (remaining=216) +0ms
  eslint:rules Loading rule 'no-trailing-spaces' (remaining=215) +0ms
  eslint:rules Loading rule 'no-whitespace-before-property' (remaining=214) +0ms
  eslint:rules Loading rule 'nonblock-statement-body-position' (remaining=213) +1ms
  eslint:rules Loading rule 'object-curly-newline' (remaining=212) +0ms
  eslint:rules Loading rule 'object-curly-spacing' (remaining=211) +0ms
  eslint:rules Loading rule 'object-property-newline' (remaining=210) +1ms
  eslint:rules Loading rule 'one-var-declaration-per-line' (remaining=209) +0ms
  eslint:rules Loading rule 'operator-linebreak' (remaining=208) +0ms
  eslint:rules Loading rule 'padded-blocks' (remaining=207) +1ms
  eslint:rules Loading rule 'quote-props' (remaining=206) +0ms
  eslint:rules Loading rule 'rest-spread-spacing' (remaining=205) +0ms
  eslint:rules Loading rule 'semi' (remaining=204) +1ms
  eslint:rules Loading rule 'semi-spacing' (remaining=203) +0ms
  eslint:rules Loading rule 'semi-style' (remaining=202) +0ms
  eslint:rules Loading rule 'space-before-blocks' (remaining=201) +0ms
  eslint:rules Loading rule 'space-before-function-paren' (remaining=200) +1ms
  eslint:rules Loading rule 'space-in-parens' (remaining=199) +0ms
  eslint:rules Loading rule 'space-infix-ops' (remaining=198) +0ms
  eslint:rules Loading rule 'space-unary-ops' (remaining=197) +1ms
  eslint:rules Loading rule 'switch-colon-spacing' (remaining=196) +0ms
  eslint:rules Loading rule 'template-curly-spacing' (remaining=195) +0ms
  eslint:rules Loading rule 'template-tag-spacing' (remaining=194) +1ms
  eslint:rules Loading rule 'unicode-bom' (remaining=193) +0ms
  eslint:rules Loading rule 'wrap-iife' (remaining=192) +0ms
  eslint:rules Loading rule 'wrap-regex' (remaining=191) +0ms
  eslint:rules Loading rule 'yield-star-spacing' (remaining=190) +1ms
  eslint:rules Loading rule 'indent-legacy' (remaining=189) +0ms
  eslint:rules Loading rule 'no-spaced-func' (remaining=188) +0ms
  eslint:rules Loading rule 'newline-after-var' (remaining=187) +2ms
  eslint:rules Loading rule 'no-console' (remaining=186) +0ms
  eslint:cascading-config-array-factory Configuration was determined: ConfigArray [ { type: 'config', name: 'DefaultIgnorePattern', filePath: '', criteria: null, env: undefined, globals: undefined, ignorePattern: IgnorePattern { patterns: [Array], basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components', loose: false }, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { type: 'config', name: '.eslintrc.js » plugin:@typescript-eslint/recommended » ./configs/base', filePath: 'node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: 'node_modules/@typescript-eslint/parser/dist/index.js', id: '@typescript-eslint/parser', importerName: '.eslintrc.js » plugin:@typescript-eslint/recommended » ./configs/base', importerPath: 'node_modules/@typescript-eslint/eslint-plugin/dist/configs/base.js' }, parserOptions: { sourceType: 'module' }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { type: 'config', name: '.eslintrc.js » plugin:@typescript-eslint/recommended » ./configs/eslint-recommended', filePath: 'node_modules/@typescript-eslint/eslint-plugin/dist/configs/eslint-recommended.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { type: 'config', name: '.eslintrc.js » plugin:@typescript-eslint/recommended » ./configs/eslint-recommended#overrides[0]', filePath: 'node_modules/@typescript-eslint/eslint-plugin/dist/configs/eslint-recommended.js', criteria: { includes: [Array], excludes: null, basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components' }, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'constructor-super': 'off', 'getter-return': 'off', 'no-const-assign': 'off', 'no-dupe-args': 'off', 'no-dupe-class-members': 'off', 'no-dupe-keys': 'off', 'no-func-assign': 'off', 'no-import-assign': 'off', 'no-new-symbol': 'off', 'no-obj-calls': 'off', 'no-redeclare': 'off', 'no-setter-return': 'off', 'no-this-before-super': 'off', 'no-undef': 'off', 'no-unreachable': 'off', 'no-unsafe-negation': 'off', 'no-var': 'error', 'prefer-const': 'error', 'prefer-rest-params': 'error', 'prefer-spread': 'error', 'valid-typeof': 'off' }, settings: undefined }, { type: 'config', name: '.eslintrc.js » plugin:@typescript-eslint/recommended', filePath: 'node_modules/@typescript-eslint/eslint-plugin/dist/index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/adjacent-overload-signatures': 'error', '@typescript-eslint/ban-ts-comment': 'error', '@typescript-eslint/ban-types': 'error', '@typescript-eslint/explicit-module-boundary-types': 'warn', 'no-array-constructor': 'off', '@typescript-eslint/no-array-constructor': 'error', 'no-empty-function': 'off', '@typescript-eslint/no-empty-function': 'error', '@typescript-eslint/no-empty-interface': 'error', '@typescript-eslint/no-explicit-any': 'warn', '@typescript-eslint/no-extra-non-null-assertion': 'error', 'no-extra-semi': 'off', '@typescript-eslint/no-extra-semi': 'error', '@typescript-eslint/no-inferrable-types': 'error', '@typescript-eslint/no-misused-new': 'error', '@typescript-eslint/no-namespace': 'error', '@typescript-eslint/no-non-null-asserted-optional-chain': 'error', '@typescript-eslint/no-non-null-assertion': 'warn', '@typescript-eslint/no-this-alias': 'error', 'no-unused-vars': 'off', '@typescript-eslint/no-unused-vars': 'warn', '@typescript-eslint/no-var-requires': 'error', '@typescript-eslint/prefer-as-const': 'error', '@typescript-eslint/prefer-namespace-keyword': 'error', '@typescript-eslint/triple-slash-reference': 'error' }, settings: undefined }, { type: 'config', name: '.eslintrc.js » plugin:react/recommended', filePath: 'node_modules/eslint-plugin-react/index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { ecmaFeatures: [Object] }, plugins: { react: [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'react/display-name': 2, 'react/jsx-key': 2, 'react/jsx-no-comment-textnodes': 2, 'react/jsx-no-duplicate-props': 2, 'react/jsx-no-target-blank': 2, 'react/jsx-no-undef': 2, 'react/jsx-uses-react': 2, 'react/jsx-uses-vars': 2, 'react/no-children-prop': 2, 'react/no-danger-with-children': 2, 'react/no-deprecated': 2, 'react/no-direct-mutation-state': 2, 'react/no-find-dom-node': 2, 'react/no-is-mounted': 2, 'react/no-render-return-value': 2, 'react/no-string-refs': 2, 'react/no-unescaped-entities': 2, 'react/no-unknown-property': 2, 'react/no-unsafe': 0, 'react/prop-types': 2, 'react/react-in-jsx-scope': 2, 'react/require-render-return': 2 }, settings: undefined }, { type: 'config', name: '.eslintrc.js » plugin:react-hooks/recommended', filePath: 'node_modules/eslint-plugin-react-hooks/index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: { 'react-hooks': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'react-hooks/rules-of-hooks': 'error', 'react-hooks/exhaustive-deps': 'warn' }, settings: undefined }, { type: 'config', name: '.eslintrc.js » eslint-config-prettier', filePath: 'node_modules/eslint-config-prettier/index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'arrow-body-style': 0, curly: 0, 'lines-around-comment': 0, 'max-len': 0, 'no-confusing-arrow': 0, 'no-mixed-operators': 0, 'no-tabs': 0, 'no-unexpected-multiline': 0, 'prefer-arrow-callback': 0, quotes: 0, 'array-bracket-newline': 'off', 'array-bracket-spacing': 'off', 'array-element-newline': 'off', 'arrow-parens': 'off', 'arrow-spacing': 'off', 'block-spacing': 'off', 'brace-style': 'off', 'comma-dangle': 'off', 'comma-spacing': 'off', 'comma-style': 'off', 'computed-property-spacing': 'off', 'dot-location': 'off', 'eol-last': 'off', 'func-call-spacing': 'off', 'function-call-argument-newline': 'off', 'function-paren-newline': 'off', 'generator-star': 'off', 'generator-star-spacing': 'off', 'implicit-arrow-linebreak': 'off', indent: 'off', 'jsx-quotes': 'off', 'key-spacing': 'off', 'keyword-spacing': 'off', 'linebreak-style': 'off', 'multiline-ternary': 'off', 'newline-per-chained-call': 'off', 'new-parens': 'off', 'no-arrow-condition': 'off', 'no-comma-dangle': 'off', 'no-extra-parens': 'off', 'no-extra-semi': 'off', 'no-floating-decimal': 'off', 'no-mixed-spaces-and-tabs': 'off', 'no-multi-spaces': 'off', 'no-multiple-empty-lines': 'off', 'no-reserved-keys': 'off', 'no-space-before-semi': 'off', 'no-trailing-spaces': 'off', 'no-whitespace-before-property': 'off', 'no-wrap-func': 'off', 'nonblock-statement-body-position': 'off', 'object-curly-newline': 'off', 'object-curly-spacing': 'off', 'object-property-newline': 'off', 'one-var-declaration-per-line': 'off', 'operator-linebreak': 'off', 'padded-blocks': 'off', 'quote-props': 'off', 'rest-spread-spacing': 'off', semi: 'off', 'semi-spacing': 'off', 'semi-style': 'off', 'space-after-function-name': 'off', 'space-after-keywords': 'off', 'space-before-blocks': 'off', 'space-before-function-paren': 'off', 'space-before-function-parentheses': 'off', 'space-before-keywords': 'off', 'space-in-brackets': 'off', 'space-in-parens': 'off', 'space-infix-ops': 'off', 'space-return-throw-case': 'off', 'space-unary-ops': 'off', 'space-unary-word-ops': 'off', 'switch-colon-spacing': 'off', 'template-curly-spacing': 'off', 'template-tag-spacing': 'off', 'unicode-bom': 'off', 'wrap-iife': 'off', 'wrap-regex': 'off', 'yield-star-spacing': 'off', 'indent-legacy': 'off', 'no-spaced-func': 'off' }, settings: undefined }, { type: 'config', name: '.eslintrc.js » eslint-config-prettier/@typescript-eslint', filePath: 'node_modules/eslint-config-prettier/@typescript-eslint.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/quotes': 0, '@typescript-eslint/brace-style': 'off', '@typescript-eslint/comma-spacing': 'off', '@typescript-eslint/func-call-spacing': 'off', '@typescript-eslint/indent': 'off', '@typescript-eslint/keyword-spacing': 'off', '@typescript-eslint/member-delimiter-style': 'off', '@typescript-eslint/no-extra-parens': 'off', '@typescript-eslint/no-extra-semi': 'off', '@typescript-eslint/semi': 'off', '@typescript-eslint/space-before-function-paren': 'off', '@typescript-eslint/type-annotation-spacing': 'off' }, settings: undefined }, { type: 'config', name: '.eslintrc.js » eslint-config-prettier/babel', filePath: 'node_modules/eslint-config-prettier/babel.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'babel/quotes': 0, 'babel/object-curly-spacing': 'off', 'babel/semi': 'off' }, settings: undefined }, { type: 'config', name: '.eslintrc.js » eslint-config-prettier/react', filePath: 'node_modules/eslint-config-prettier/react.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'react/jsx-child-element-spacing': 'off', 'react/jsx-closing-bracket-location': 'off', 'react/jsx-closing-tag-location': 'off', 'react/jsx-curly-newline': 'off', 'react/jsx-curly-spacing': 'off', 'react/jsx-equals-spacing': 'off', 'react/jsx-first-prop-new-line': 'off', 'react/jsx-indent': 'off', 'react/jsx-indent-props': 'off', 'react/jsx-max-props-per-line': 'off', 'react/jsx-one-expression-per-line': 'off', 'react/jsx-props-no-multi-spaces': 'off', 'react/jsx-tag-spacing': 'off', 'react/jsx-wrap-multilines': 'off', 'react/jsx-space-before-closing': 'off' }, settings: undefined }, { type: 'config', name: '.eslintrc.js', filePath: '.eslintrc.js', criteria: null, env: { browser: true, node: true, es2020: true }, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: 'node_modules/@typescript-eslint/parser/dist/index.js', id: '@typescript-eslint/parser', importerName: '.eslintrc.js', importerPath: '.eslintrc.js' }, parserOptions: { ecmaFeatures: [Object], ecmaVersion: 11, sourceType: 'module' }, plugins: { react: [Object], 'react-hooks': [Object], '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/no-unused-vars': [Array], '@typescript-eslint/no-use-before-define': [Array], '@typescript-eslint/no-inferrable-types': 'off', 'newline-after-var': [Array], 'no-console': 'warn', 'no-undef': 'error', 'react/display-name': 'off', 'react/jsx-fragments': [Array], 'react/prop-types': [Array] }, settings: { react: [Object] } }, { type: 'config', name: '.eslintrc.js#overrides[0]', filePath: '.eslintrc.js', criteria: { includes: [Array], excludes: null, basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components' }, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/explicit-module-boundary-types': 'off' }, settings: undefined }, { type: 'config', name: '.eslintrc.js#overrides[1]', filePath: '.eslintrc.js', criteria: { includes: [Array], excludes: null, basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components' }, env: { jest: true }, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/no-namespace': 'off', '@typescript-eslint/no-explicit-any': 'off', '@typescript-eslint/no-var-requires': 'off', '@typescript-eslint/explicit-function-return-type': 'off', 'no-console': 'off' }, settings: undefined }, { type: 'config', name: '.eslintrc.js#overrides[2]', filePath: '.eslintrc.js', criteria: { includes: [Array], excludes: null, basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components' }, env: { node: true }, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/no-explicit-any': 'off', 'no-console': 'off' }, settings: undefined }, { type: 'config', name: '.eslintrc.js#overrides[3]', filePath: '.eslintrc.js', criteria: { includes: [Array], excludes: null, basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components' }, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/no-var-requires': 'off', 'no-console': 'off' }, settings: undefined } ] on src +52ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components', loose: false } ] +451ms
  eslint:ignore-pattern   processed: { basePath: '/Users/xh/Workspace/development-docker-setup/webapp/gauss/app/react-components', patterns: [ '/**/node_modules/*' ] } +0ms
  eslint:ignore-pattern Check { filePath:
   'src/test.ts',
  dot: false,
  relativePath: 'src/test.ts',
  result: false } +1ms
  eslint:cli-engine Lint src/test.ts +0ms
  eslint:linter Linting code for src/test.ts (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: src/test.ts +0ms
  eslint:linter Generating fixed text for src/test.ts (pass 1) +57ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:file-enumerator Complete iterating files: ["src/test.ts"] +508ms
  eslint:cli-engine Linting complete in: 509ms +57ms

src/test.ts
  2:22  warning  'foobar' is assigned a value but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (0 errors, 1 warning)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.1` |
| `@typescript-eslint/parser`        | `3.7.1` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.5.0` |
| `node`                             | `10.20.0` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
Please raise an issue with the base ESLint repo. Our rule just extends the base functionality.
https://github.com/eslint/eslint
